src/playbacks/html5_video/html5_video.js

```diff
@@ -166,51 +196,65 @@ export default class HTML5Video extends Playback {
     return this.isReadyState
   }
 
-  playing() {
-    this.trigger(Events.PLAYBACK_PLAY);
+  determineIfPlayheadMoving() {
+    var before = this.playheadMovingTimeOnCheck
+    var now = this.el.currentTime
+    this.playheadMoving = before !== now
+    this.playheadMovingTimeOnCheck = now
+    this.handleBufferingEvents()
   }
 
-  paused() {
-    this.trigger(Events.PLAYBACK_PAUSE);
+  onLoadStart() {
+    if (this.el.preload !== 'none') {
+      // when preload is none the onLoadStart event is still fired
+      // immediately. Pretend that load starts in play()
+      this.loadStarted = true
+      this.handleBufferingEvents()
+    }
   }
 
-  onEnded() {
-    this.trigger(Events.PLAYBACK_BUFFERFULL, this.name)
-    this.trigger(Events.PLAYBACK_ENDED, this.name)
-    this.trigger(Events.PLAYBACK_TIMEUPDATE, { current: 0, total: this.el.duration }, this.name)
+  onPlaying() {
+    this.handleBufferingEvents()
+    this.trigger(Events.PLAYBACK_PLAY)
   }
 
-  stalled() {
-    if (this.getPlaybackType() === Playback.VOD && this.el.readyState < this.el.HAVE_FUTURE_DATA) {
-      this.trigger(Events.PLAYBACK_BUFFERING, this.name)
-    }
+  onPause() {
+    this.handleBufferingEvents()
+    this.trigger(Events.PLAYBACK_PAUSE)
   }
 
-  waiting() {
-    if(this.el.readyState < this.el.HAVE_FUTURE_DATA) {
-      this.trigger(Events.PLAYBACK_BUFFERING, this.name)
-    }
+  onEnded() {
+    this.handleBufferingEvents()
+    this.trigger(Events.PLAYBACK_ENDED, this.name)
   }
 
-  bufferFull() {
-    if (this.options.poster && this.firstBuffer) {
-      this.firstBuffer = false
-      if (!this.isPlaying()) {
-        this.el.poster = this.options.poster
+  // The playback should be classed as buffering if the following are true:
+  // - the ready state is less then HAVE_FUTURE_DATA or the playhead isn't moving and it should be
+  // - the media hasn't "ended"
+  // - loading has started
+  handleBufferingEvents() {
+    var playheadShouldBeMoving = !this.el.ended && !this.el.paused
+    var buffering = this.loadStarted && !this.el.ended && ((playheadShouldBeMoving && !this.playheadMoving) || this.el.readyState < this.el.HAVE_FUTURE_DATA)
+    if (this.bufferingState !== buffering) {
+      this.bufferingState = buffering
+      if (buffering) {
+        this.trigger(Events.PLAYBACK_BUFFERING, this.name)
+      }
+      else {
+        this.trigger(Events.PLAYBACK_BUFFERFULL, this.name)
       }
-    } else {
-      this.el.poster = ''
     }
-    this.trigger(Events.PLAYBACK_BUFFERFULL, this.name)
   }
 
-  error(event) {
+  onError(event) {
     this.trigger(Events.PLAYBACK_ERROR, this.el.error, this.name)
   }
 
   destroy() {
     this.stop()
     this.el.src = ''
+    this.src = null
+    clearInterval(this.playheadMovingTimer)
```

Comment:
Maybe we should `clearInterval` at stop too otherwise it'll keep checking bufferstate... this would require to start it again on play (if it's not already started).

I'm thinking that after a media is done, it'll keep triggering `BUFFERFULL`